### PR TITLE
Add collection to $casts

### PR DESCRIPTION
**Pull request by cliffordjames**
Add support for collection to $casts

### CODE REVIEW

File: src/Illuminate/Database/Eloquent/Model.php [hunk 1]
```diff
@@ -2739,6 +2739,8 @@ protected function castAttribute($key, $value)
 			case 'array':
 			case 'json':
 				return json_decode($value, true);
+			case 'collection':
+				return new Collection(json_decode($value, true));
```
Comment (JosephSilber) [inline src/Illuminate/Database/Eloquent/Model.php]: Use `$this->newCollection`, so that it'll use the user-provided collection class.

**Comment by cliffordjames** [comment]
I've updated the pull request!

**Comment by taylorotwell** [comment]
Seems like this should be listed in `isJsonCastable`. How are you intending collections to be stored? I assume the same as the `array` or `json` casts?

**Comment by cliffordjames** [comment]
Yes thats the idea, same as the `array` or `json` casts.

File: src/Illuminate/Database/Eloquent/Model.php [hunk 2]
```diff
@@ -2691,8 +2691,10 @@ protected function isJsonCastable($key)
 		if ($this->hasCast($key))
 		{
 			$type = $this->getCastType($key);
-
-			return $type === 'array' || $type === 'json' || $type === 'object';
+			
+            $jsonCastables = array('array', 'json', 'object', 'collection');
```
Comment (GrahamCampbell) [inline src/Illuminate/Database/Eloquent/Model.php]: Indentation is wrong.

**Pull request merged**